UnitTests/GitUI.Tests/UserControls/BlameControlTests.cs

```diff
@@ -149,5 +150,105 @@ public void BuildBlameContents_WithDateButNotTime()
+            {
+                DateTime lineDate = lineDates[index];
+                yield return new GitBlameLine(
+                    new GitBlameCommit(null, "Author1", "@Author1", lineDate, string.Empty,
+                        "Commiter", "@Committer", lineDate, string.Empty, "Summary1", "file"),
+                    index + 1, index + 1, "text");
+            }
+        }
+
+        [Test]
+        public void CalculateBlameGutterData_When_date_is_older_than_artificial_old_boundary_Then_it_define_first_age_bucket()
+        {
+            // Given
+            IReadOnlyList<GitBlameLine> blameLines = CreateBlameLine(new DateTime(2008, 9, 1)).ToList();
+
+            // When
+            var blameEntries = _blameControl.GetTestAccessor().CalculateBlameGutterData(blameLines);
+
+            // Then
+            blameEntries.Should().HaveCount(1);
+            blameEntries[0].AgeBucketIndex.Should().Be(0);
+        }
+
+        [Test]
+        public void CalculateBlameGutterData_When_date_is_newer_than_artificial_old_boundary_Then_it_falls_in_a_later_age_bucket()
+        {
+            // Given
+            IReadOnlyList<GitBlameLine> blameLines = CreateBlameLine(DateTime.Now.AddMonths(-18)).ToList();
+
+            // When
+            var blameEntries = _blameControl.GetTestAccessor().CalculateBlameGutterData(blameLines);
+
+            // Then
+            blameEntries.Should().HaveCount(1);
+            blameEntries[0].AgeBucketIndex.Should().Be(3);
+        }
+
+        [Test]
+        public void CalculateBlameGutterData_When_date_is_DateMin_Then_it_falls_in_a_first_age_bucket()
+        {
+            // Given
+            IReadOnlyList<GitBlameLine> blameLines = CreateBlameLine(DateTime.Now.AddMonths(-18), DateTime.MinValue).ToList();
+
+            // When
+            var blameEntries = _blameControl.GetTestAccessor().CalculateBlameGutterData(blameLines);
+
+            // Then
+            blameEntries.Should().HaveCount(2);
+            blameEntries[1].AgeBucketIndex.Should().Be(0);
+        }
+
+        [Test]
+        public void CalculateBlameGutterData_When_all_dates_are_DateMin_Then_they_falls_in_a_first_age_bucket()
+        {
+            // Given
+            IReadOnlyList<GitBlameLine> blameLines = CreateBlameLine(DateTime.MinValue, DateTime.MinValue).ToList();
+
+            // When
+            var blameEntries = _blameControl.GetTestAccessor().CalculateBlameGutterData(blameLines);
+
+            // Then
+            blameEntries.Should().HaveCount(2);
+            blameEntries[0].AgeBucketIndex.Should().Be(0);
+            blameEntries[1].AgeBucketIndex.Should().Be(0);
+        }
+
+        [Test]
+        public void CalculateBlameGutterData_When_dates_just_after_age_bucket_limit_Then_One_date_in_each_age_bucket()
+        {
+            // Given
+            var now = DateTime.Now;
+            var marginError = TimeSpan.FromMilliseconds(10); // Due to the DateTime.Now value which is slightly different
```

Comment:
Can it be hours instead of milliseconds for better reliability?

Reply:
Sorry, I forgot to fix this one because I didn't know immediately what to think about your comment...

Using milliseconds better express the idea that the calculation is exact (except for a small time corresponding to the time between this `DateTime.Now` and the one in the tested method, so just the time needed to generate the values). I imagine that 10ms should be enough...

But indeed, you're right, I also want the test to be reliable.
So, what do you think if I add 1 second? I don't like the idea to add **hours**...

Reply:
can we have a deal at 10 minutes? :)